scripts/src/java/org/oppia/android/scripts/common/BazelClient.kt

```diff
@@ -130,6 +131,59 @@ class BazelClient(private val rootDirectory: File, private val commandExecutor:
     return correctedTargets
   }
 
+  /**
+   * Runs code coverage for the specified Bazel test target
+   *
+   * @param bazelTestTarget Bazel test target for which code coverage will be run
+   * @return the generated coverage data as a list of strings
+   * or null if the coverage data file could not be parsed
+   *
+   * Null return typically occurs when the coverage command fails to generate the 'coverage.dat' file
+   * This can happen due to: Test failures or misconfigurations that prevent the coverage data
+   * from being generated properly
+   */
+  fun runCoverageForTestTarget(bazelTestTarget: String): List<String>? {
+    val coverageData = executeBazelCommand(
+      "coverage",
+      bazelTestTarget
+    )
+    return parseCoverageDataFile(coverageData)?.let { path ->
+      readDatFile(path)
+    }
+  }
+
+  /**
+   * Parse the coverage command result to extract the path of the coverage data file.
+   *
+   * @param data the result from the execution of the coverage command
+   * @return the extracted path of the coverage data file.
+   */
+  private fun parseCoverageDataFile(data: List<String>): String? {
+    val regex = ".*coverage\\.dat$".toRegex()
+    for (line in data) {
+      val match = regex.find(line)
+      val extractedPath = match?.value?.substringAfterLast(",")?.trim()
+      if (extractedPath != null) {
+        println("Parsed Coverage Data File: $extractedPath")
+        return extractedPath
+      }
+    }
+    return null
+  }
+
+  /**
+   * Reads the content of the .dat file as a list of lines.
+   *
+   * @param filePath path to the .dat file
+   * @return content of the .dat file as a list of strings
+   */
+  private fun readDatFile(filePath: String?): List<String>? {
```

Comment:
``filePath`` is never passed in as null, so this can be simplified.

Reply:
Done, Thanks!